File: 프로그래머스/2/42584. 주식가격/주식가격.py
from collections import deque

def solution(prices):
    answer = []
    queue = deque(prices)
    
    while queue:
        current_price = queue.popleft()
        
        time = 0
        for price in queue:
            time += 1
            if current_price > price:
                break
        answer.append(time)
    
    return answer

File: 프로그래머스/2/42584. 주식가격/주식가격.js
function solution(prices) {
    const answer = [];
    
    for (let i = 0; i < prices.length; i++) {
        const currPrice = prices[i];
        let sec = 0;
        for (let j = i + 1; j < prices.length; j++) {
            sec++;
            if (prices[j] < currPrice) break;
        }
        answer.push(sec);
    }
    
    return answer;
}